src/middleware/packages/triplestore/actions/query.js

```diff
@@ -27,8 +27,11 @@ module.exports = {
     const dataset = ctx.params.dataset || ctx.meta.dataset || this.settings.mainDataset;
 
     if (!dataset) throw new Error(`No dataset defined for triplestore query: ${query}`);
-    if (!(await ctx.call('triplestore.dataset.exist', { dataset })))
-      throw new Error(`The dataset ${dataset} doesn't exist`);
+    if (!(await ctx.call('triplestore.dataset.exist', { dataset }))) {
+      const error = new Error(`The dataset ${dataset} doesn't exist`);
+      error.code = 404;
+      throw error;
+    }
```

Comment:
It's better not to include HTTP error codes here, as it is used directly by many services. If an account doesn't exist, the `auth.account.findByUsername` action that you called in the InboxService will return nothing, and so you can throw the 404 error from there.

Reply:
fixed.